core/src/stores/block_information_store.rs

```diff
@@ -164,4 +166,14 @@ impl BlockInformationStore {
             None => (false, latest_block.slot),
         }
     }
+
+    pub fn get_block_info_by_slot(&self, slot: u64) -> Option<BlockInformation> {
+        self.blocks.iter().find_map(|iter| {
+            if iter.slot == slot {
```

Comment:
pls provide a comment, why that approach is fast enough like "fast enough because we are iterating only a small map of around 100 entries"